kythe/typescript/indexer.ts

```diff
@@ -363,16 +374,43 @@ class Vistor {
   }
 
   /** getSymbolName computes the VName (and signature) of a ts.Symbol. */
-  getSymbolName(sym: ts.Symbol, ns: TSNamespace): VName {
+  getSymbolName(sym: ts.Symbol, ns: TSNamespace, context?: ts.Node): VName {
     let vnames = this.symbolNames.get(sym);
-    if (vnames && vnames[ns]) return vnames[ns]!;
+    let declarations = sym.declarations;
+    if (!(context && declarations.length > 1) && vnames && vnames[ns])
+      return vnames[ns]!;
+    // TODO: update symbolNames table to account for context kind
 
-    if (!sym.declarations || sym.declarations.length < 1) {
+    if (!declarations || declarations.length < 1) {
       throw new Error('TODO: symbol has no declarations?');
     }
-    // TODO: think about symbols with multiple declarations.
+    // Symbols with multiple declarations are disambiguated by the context
+    // they are used in.
+    if (context && declarations.length > 1) {
+      switch (context.kind) {
+        case ts.SyntaxKind.GetAccessor:
+        case ts.SyntaxKind.SetAccessor:
+          declarations =
+              declarations.filter(decl => decl.kind === context.kind);
+          break;
+        case ts.SyntaxKind.BinaryExpression:
+          const ctx = context as ts.BinaryExpression;
+          // If the node is being assigned to it cannot reference a getter.
+          if (this.getSymbolAtLocation(ctx.left) === sym &&
+              ctx.operatorToken.kind === ts.SyntaxKind.EqualsToken) {
+            declarations = declarations.filter(
+                decl => decl.kind !== ts.SyntaxKind.GetAccessor);
+          }
+          break;
+        default:
+          // If the node is not being assigned to it cannot reference a setter.
+          declarations = declarations.filter(
+              decl => decl.kind !== ts.SyntaxKind.SetAccessor);
+          break;
+      }
+    }
 
-    const decl = sym.declarations[0];
+    const decl = declarations[0];
```

Comment:
It's still possible for declarations.length > 1 here, right?

Reply:
Yep. The approach I thought was a "best-effort" kind of thing; the role of the context is to reduce the declarations as much as possible.